clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesDataProviderIntegrationTest.java

```diff
@@ -0,0 +1,1117 @@
+
+    Set<KubernetesV2Cluster> clusters = results.get(ACCOUNT_NAME);
+    assertThat(clusters).hasSize(2);
+
+    assertThat(clusters)
+        .extracting(KubernetesV2Cluster::getName)
+        .containsExactlyInAnyOrder("service backendlb", "replicaSet backend");
+
+    Map<String, KubernetesV2Cluster> clusterLookup =
+        clusters.stream().collect(toImmutableMap(KubernetesV2Cluster::getName, c -> c));
+
+    assertBackendServiceCluster(softly, clusterLookup.get("service backendlb"));
+    assertBackendCluster(softly, clusterLookup.get("replicaSet backend"), true);
+  }
+
+  @Test
+  void getServerGroup(SoftAssertions softly) {
+    KubernetesV2ServerGroup serverGroup =
+        clusterProvider.getServerGroup(ACCOUNT_NAME, "backend-ns", "replicaSet backend-v014");
+    assertThat(serverGroup).isNotNull();
+    assertBackendPriorServerGroup(softly, serverGroup);
+  }
+
+  @Test
+  void getServerGroupWrongNamespace(SoftAssertions softly) {
+    KubernetesV2ServerGroup serverGroup =
+        clusterProvider.getServerGroup(ACCOUNT_NAME, "frontend-ns", "replicaSet backend-v014");
+    assertThat(serverGroup).isNull();
+  }
+
+  @Test
+  void getServerGroupWithDetails(SoftAssertions softly) {
+    KubernetesV2ServerGroup serverGroup =
+        clusterProvider.getServerGroup(ACCOUNT_NAME, "backend-ns", "replicaSet backend-v014", true);
+    assertThat(serverGroup).isNotNull();
+    assertBackendPriorServerGroup(softly, serverGroup);
+  }
+
+  @Test
+  void getServerGroupWithoutDetails(SoftAssertions softly) {
+    KubernetesV2ServerGroup serverGroup =
+        clusterProvider.getServerGroup(
+            ACCOUNT_NAME, "backend-ns", "replicaSet backend-v014", false);
+    assertThat(serverGroup).isNotNull();
+    // Looks like we ignore the includeDetails flag, so this is the same serverGroup as when we do
+    // include details.
+    assertBackendPriorServerGroup(softly, serverGroup);
+  }
+
+  @Test
+  void getApplicationsUnexpanded(SoftAssertions softly) {
+    Set<? extends Application> result = applicationProvider.getApplications(false);
+    softly.assertThat(result).hasSize(2);
+
+    Map<String, KubernetesV2Application> applicationLookup =
+        result.stream()
+            .collect(toImmutableMap(Application::getName, a -> (KubernetesV2Application) a));
+
+    KubernetesV2Application frontendApplication = applicationLookup.get("frontendapp");
+    softly.assertThat(frontendApplication).isNotNull();
+    if (frontendApplication != null) {
+      assertFrontendApplication(softly, frontendApplication);
+    }
+
+    KubernetesV2Application backendApplication = applicationLookup.get("backendapp");
+    softly.assertThat(frontendApplication).isNotNull();
+    if (frontendApplication != null) {
+      assertBackendApplication(softly, backendApplication);
+    }
+  }
+
+  @Test
+  void getApplicationsExpanded(SoftAssertions softly) {
+    // This is the same as the unexpanded test, as it seems like we ignore the flag.
```

Comment:
Any context on why we even have these flags? Curious if other cloud providers use them.

Reply:
Good question; I'd imagine that some other providers do (or at least one) if they were added to the interface, but I'm actually not sure.  This could also be fertile ground for performance improvements if it turns out that we're returning way more data than we need because we don't respect these flags.